### PR TITLE
SQLite - New types handled: Hex, Packed

### PR DESCRIPTION
**Pull request by Zeatynis**
Hi, I needed to handle additional types of data - Hex and Packed. Please check if I did it correctly :)

### CODE REVIEW

File: packages/transpiler/src/db/sqlite_database_schema.ts [hunk 1]
```diff
@@ -52,8 +52,13 @@ export class SQLiteDatabaseSchema {
     } else if (type instanceof abaplint.BasicTypes.XStringType) {
       // it will be fine, the runtime representation of xstring is also text
       return `TEXT`;
+    } else if (type instanceof abaplint.BasicTypes.HexType) {
+      // it will be fine, the runtime representation of hex is also text
+      return `TEXT`;
```
Comment (larshp) [inline packages/transpiler/src/db/sqlite_database_schema.ts:57]: return `NCHAR(${type.getLength() * 2})`;

Comment (larshp) [reply packages/transpiler/src/db/sqlite_database_schema.ts:57]: hex types have a length, storing any length text in databases is expensive

File: packages/transpiler/src/db/sqlite_database_schema.ts [hunk 2]
```diff
@@ -52,8 +52,13 @@ export class SQLiteDatabaseSchema {
...
     } else if (type instanceof abaplint.BasicTypes.IntegerType) {
       return `INT`;
+    } else if (type instanceof abaplint.BasicTypes.PackedType){
+      return `DECIMAL(${type.getLength()},${type.getDecimals()})`;
```
Comment (larshp) [inline packages/transpiler/src/db/sqlite_database_schema.ts:61]: guess this will be okay for now, but I'm not sure if it works

Comment (Zeatynis) [reply packages/transpiler/src/db/sqlite_database_schema.ts:60]: ye, me neither. I don't know how to check it 🤣

Comment (larshp) [reply packages/transpiler/src/db/sqlite_database_schema.ts:60]: write a unit test, https://github.com/abaplint/transpiler/blob/main/test/database.ts

**Comment by larshp** [comment]
published in 2.0.51

**Pull request merged**